packages/cli-platform-apple/src/commands/runCommand/getBuildPath.ts

```diff
@@ -42,63 +19,15 @@ export async function getBuildPath(
     throw new CLIError('Failed to get the app name.');
   }
 
-  return `${targetBuildDir}${
-    isCatalyst ? '-maccatalyst' : ''
-  }/${executableFolderPath}`;
-}
-
-async function getTargetPaths(
-  buildSettings: string,
-  scheme: string,
-  target: string | undefined,
-) {
-  const settings = JSON.parse(buildSettings);
-
-  const targets = settings.map(
-    ({target: settingsTarget}: any) => settingsTarget,
-  );
-
-  let selectedTarget = targets[0];
-
-  if (target) {
-    if (!targets.includes(target)) {
-      logger.info(
-        `Target ${chalk.bold(target)} not found for scheme ${chalk.bold(
-          scheme,
-        )}, automatically selected target ${chalk.bold(selectedTarget)}`,
-      );
-    } else {
-      selectedTarget = target;
-    }
-  }
-
-  // Find app in all building settings - look for WRAPPER_EXTENSION: 'app',
-
-  const targetIndex = targets.indexOf(selectedTarget);
-
-  const wrapperExtension =
-    settings[targetIndex].buildSettings.WRAPPER_EXTENSION;
-
-  if (wrapperExtension === 'app') {
-    return {
-      targetBuildDir: settings[targetIndex].buildSettings.TARGET_BUILD_DIR,
-      executableFolderPath:
-        settings[targetIndex].buildSettings.EXECUTABLE_FOLDER_PATH,
-    };
+  if (!fullProductName) {
+    throw new CLIError('Failed to get product name.');
   }
 
-  return {};
-}
-
-function getPlatformName(buildOutput: string) {
-  // Xcode can sometimes escape `=` with a backslash or put the value in quotes
-  const platformNameMatch = /export PLATFORM_NAME\\?="?(\w+)"?$/m.exec(
-    buildOutput,
-  );
-  if (!platformNameMatch) {
-    throw new CLIError(
-      'Couldn\'t find "PLATFORM_NAME" variable in xcodebuild output. Please report this issue and run your project with Xcode instead.',
-    );
+  if (platform === 'macos') {
+    return path.join(targetBuildDir, fullProductName);
+  } else if (isCatalyst) {
+    return path.join(targetBuildDir, '-maccatalyst', executableFolderPath);
```

Comment:
Should `isCatalyst` be checked first? What's the value of `platform` when we're targeting Catalyst?